optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolver.java

```diff
@@ -109,11 +109,17 @@ public Score getBestScore() {
 
     @Override
     public long getTimeMillisSpent() {
```

Comment:
DefaultSolverScope already contains methods `calculateTimeMillisSpentUpToNow` and `getTimeMillisSpent`. Why not to reuse them?

Reply:
Because calculateTimeMillisSpentUpToNow() isn't multi-thread safe. It is only solver-thread safe.
In a multi-threaded environment, start might be null, end might be null or might not.
And restarts might make a non-null end, null again.